network/dnscrypt-proxy/dnscrypt-proxy.SlackBuild

```diff
@@ -22,101 +23,87 @@
+    bailout
 fi
 
-set -e
-
 rm -rf $PKG
-mkdir -p $TMP $PKG $OUTPUT
+mkdir -p $TMP/$PRGNAM-$VERSION $PKG $OUTPUT
 cd $TMP
 rm -rf $PRGNAM-$VERSION
-mkdir -p $PRGNAM-$VERSION/src/$DOMAIN/$ORG
-cd $PRGNAM-$VERSION/src/$DOMAIN/$ORG
-tar xvf $CWD/$PRGNAM-$VERSION.tar.gz
-mv $PRGNAM-$VERSION $PRGNAM
-cd $TMP/$PRGNAM-$VERSION
+
+if [ $ARCH == i586 ]; then
+    tar xvf $CWD/$PRGNAM-linux_i386-$VERSION.tar.gz --transform="s/linux-i386/$PRGNAM-$VERSION/"
+elif [ $ARCH == x86_64 ]; then
+    tar xvf $CWD/$PRGNAM-linux_x86_64-$VERSION.tar.gz --transform="s/linux-x86_64/$PRGNAM-$VERSION/"
+fi
+
+cd $PRGNAM-$VERSION
 chown -R root:root .
-find -L . \
+
+find -L . $CWD \
  \( -perm 777 -o -perm 775 -o -perm 750 -o -perm 711 -o -perm 555 \
   -o -perm 511 \) -exec chmod 755 {} \; -o \
  \( -perm 666 -o -perm 664 -o -perm 640 -o -perm 600 -o -perm 444 \
   -o -perm 440 -o -perm 400 \) -exec chmod 644 {} \;
 
-export GO111MODULE=auto
-export GOPATH="$TMP/$PRGNAM-$VERSION"
-
-go install -a -x ./...
-
-cd $TMP/$PRGNAM-$VERSION/src/$DOMAIN/$ORG/$PRGNAM
-
-mkdir -p $PKG/usr/sbin
-
-install -m 755 "$TMP/$PRGNAM-$VERSION/bin/${PRGNAM}" $PKG/usr/sbin/${PRGNAM}
-
-find $PKG -print0 | xargs -0 file | grep -e "executable" -e "shared object" | grep ELF \
-  | cut -f 1 -d : | xargs strip --strip-unneeded 2> /dev/null || true
-
-mkdir -p $PKG/var/log/$PRGNAM $PKG/etc/default $PKG/etc/$PRGNAM
-chmod 0700 $PKG/var/log/$PRGNAM
-sed "s/@VERSION@/$VERSION/" $CWD/$PRGNAM.default > $PKG/etc/default/$PRGNAM.new
-install -D -m 0755 $CWD/rc.$PRGNAM $PKG/etc/rc.d/rc.$PRGNAM.new
-install -D -m 0644 $CWD/$PRGNAM.toml $PKG/etc/$PRGNAM/$PRGNAM.toml.new
+mkdir -p $PKG/usr/{bin,doc/$PRGNAM-$VERSION} \
+    $PKG/etc/{$PRGNAM,rc.d} \
+    $PKG/var/{log,run}/$PRGNAM \
+    $PKG/install
 
-mkdir -p $PKG/usr/doc/$PRGNAM-$VERSION
-cp -a \
-  ChangeLog LICENSE README.md utils $PRGNAM/example-* \
-  $PKG/usr/doc/$PRGNAM-$VERSION
-sed "s/@VERSION@/$VERSION/g" $CWD/README.Slackware > $PKG/usr/doc/$PRGNAM-$VERSION/README.Slackware
-cat $CWD/dnsmasq.conf > $PKG/usr/doc/$PRGNAM-$VERSION/dnsmasq.conf
-cat $CWD/named.conf > $PKG/usr/doc/$PRGNAM-$VERSION/named.conf
-cat $CWD/$PRGNAM.SlackBuild > $PKG/usr/doc/$PRGNAM-$VERSION/$PRGNAM.SlackBuild
+find . $CWD -mindepth 1 -type f \( -name '*.txt' -o -name '*.toml' -o -name '*.pem' \) -exec cp -a {} $PKG/etc/$PRGNAM/ \; -o \
+    \( -name 'LICENSE' \) -exec cp -a {} $PKG/usr/doc/$PRGNAM-$VERSION/ \; -o \
+    \( -name 'README.Slackware' \) -exec cp {} $PKG/usr/doc/$PRGNAM-$VERSION/ \; -o \
+    \( -name "$PRGNAM.SlackBuild" \) -exec cp {} $PKG/usr/doc/$PRGNAM-$VERSION/ \; -o \
+    \( -name "$PRGNAM" \) -exec cp -a {} $PKG/usr/bin/ \; -o \
+    \( -name "$PRGNAM.toml" \) -exec cp {} $PKG/etc/$PRGNAM/ \; -o \
+    \( -name "rc.$PRGNAM" \) -exec cp {} $PKG/etc/rc.d/ \; -o \
+    \( -name 'slack-desc' \) -exec cp {} $PKG/install/ \;
 
-mkdir -p $PKG/install
-cat $CWD/slack-desc > $PKG/install/slack-desc
 cat $CWD/doinst.sh > $PKG/install/doinst.sh
```

Comment:
I appreciate your submission and your updating this build, especially since we have no updates from T3slider for ~3 year now, but sorry, please read this page for notes:
https://slackbuilds.org/guidelines/

Especially this part:

> Follow our template scripts as much as possible. Deviations are fine where needed, but don't change things to be "more clever" or some such - we like consistency

We have something like 9000 scripts in the repo and yours is going to be an outlier if we approve it in this way. It is much simpler if everyone uses a similar pattern, even if there are other ways to do it.

Reply:
> I appreciate your submission and your updating this build, especially since we have no updates from T3slider for ~3 year now, but sorry, please read this page for notes: https://slackbuilds.org/guidelines/
> 
> Especially this part:
> 
> > Follow our template scripts as much as possible. Deviations are fine where needed, but don't change things to be "more clever" or some such - we like consistency
> 
> We have something like 9000 scripts in the repo and yours is going to be an outlier if we approve it in this way. It is much simpler if everyone uses a similar pattern, even if there are other ways to do it.

I will be updating the script to reflect your remarks and have it fall in line with the SBo template. I appreciate your feedback :)